mne/annotations.py

```diff
@@ -817,10 +817,18 @@ def _select_annotations_based_on_description(descriptions, event_id, regexp):
 
 
 def _check_event_id(event_id, raw):
+    from mne.io.brainvision.brainvision import _BVEventParser
+    from mne.io.brainvision.brainvision import _check_bv_annot
+
     if event_id is None:
         return _DefaultEventParser()
     elif event_id == 'auto':
-        return getattr(raw, '_get_auto_event_id', _DefaultEventParser)()
+        candidate = getattr(raw, '_get_auto_event_id', _DefaultEventParser)
+        if candidate == _DefaultEventParser and \
+           _check_bv_annot(raw.annotations.description):
+            return _BVEventParser()
```

Comment:
There was an additional qualifier @agramfort wanted which was that `raw` be an instance of `RawFIF` (or I think it should also include `RawArray`)

Reply:
That's why I started by implementing this inside a class method of `Raw`, but then `getattr` does nothing because raw is a subclass and its always present

Reply:
up to some point, this part was easy: just get read of the default and the base class would already give you `_DefaultEventParser`.

But the method itself was weird:
```
    @classmethod
    def _get_auto_event_id(cls):
        """Return default ``event_id`` behavior unless it matches Brainvision."""
        if _check_annotation_descriptions:  # self is not present
            return _DefaultEventParser()
        else:
            return _BVEventParser()  # return a free helper function of a class that inherates from RaW.
```

returning `_BVEventParser` is not a problem perse, but it breaks the conceptual structure.

Reply:
you should be able to default to _BVEventParser() only if you have a RawBrainVision, RawFIF or RawArray

Reply:
If that's the contract then we need to create `RawFiF` which we don't have.
```py
class A:
    pass
class B(A):
    pass
a = A()
b = B()
print(isinstance(a, A), isinstance(b,A), issubclass(type(b),A), issubclass(type(a),A))
```
```sh
True True True True
```